label_studio/tasks/serializers.py

```diff
@@ -162,6 +164,13 @@ def to_representation(self, instance):
             if self.context.get('resolve_uri', False):
                 instance.data = instance.resolve_uri(instance.data, proxy=self.context.get('proxy', False))
 
+            # resolve S3 URLs
```

Comment:
We already have a generic URL resolving function on the previous line. And it is also call `resolve_s3_url`. So in some cases, we will get double URL resolving. You can fix the logic of retrieving URL from `instance.data` there: https://github.com/wgillett/label-studio/blob/irobot-2021-11-30/label_studio/io_storages/utils.py#L12